python/semantic_kernel/connectors/ai/ollama/services/ollama_text_completion.py

```diff
@@ -0,0 +1,91 @@
+)
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class OllamaTextCompletion(TextCompletionClientBase, AIServiceClientBase):
+    """
+    Initializes a new instance of the OllamaTextCompletion class.
+
+    Make sure to have the ollama service running either locally or remotely.
+
+    Arguments:
+        ai_model_id {str} -- Ollama model name, see https://ollama.ai/library
+        url {Optional[Union[str, HttpUrl]]} -- URL of the Ollama server, defaults to http://localhost:11434/api/generate
+    """
+
+    url: HttpUrl = "http://localhost:11434/api/generate"
+    session: Optional[aiohttp.ClientSession] = None
+
+    async def complete_async(
+        self,
+        prompt: str,
+        request_settings: OllamaTextRequestSettings,
+        **kwargs,
+    ) -> Union[str, List[str]]:
+        request_settings.prompt = prompt
+        request_settings.stream = False
+        if self.session:
+            async with self.session.post(self.url, json=request_settings.prepare_settings_dict()) as response:
+                response.raise_for_status()
+                return await response.text()
+        async with aiohttp.ClientSession() as session:
+            async with session.post(self.url, json=request_settings.prepare_settings_dict()) as response:
+                response.raise_for_status()
+                return await response.text()
+
+    async def complete_stream_async(
+        self,
+        prompt: str,
+        request_settings: OllamaTextRequestSettings,
+        **kwargs,
+    ):
+        """
+        Streams a text completion using a Hugging Face model.
+        Note that this method does not support multiple responses.
+
+        Arguments:
+            prompt {str} -- Prompt to complete.
+            request_settings {HuggingFaceRequestSettings} -- Request settings.
+
+        Yields:
+            str -- Completion result.
+        """
+        request_settings.prompt = prompt
+        request_settings.stream = True
+        if self.session:
+            async with self.session.post(self.url, json=request_settings.prepare_settings_dict()) as response:
+                response.raise_for_status()
+                async for line in response.content:
+                    body = json.loads(line)
+                    response_part = body.get("response")
+                    yield response_part
+                    if body.get("done"):
+                        break
+        if not self.session:
+            async with aiohttp.ClientSession() as session:
+                async with session.post(self.url, json=request_settings.prepare_settings_dict()) as response:
+                    response.raise_for_status()
+                    async for line in response.content:
+                        body = json.loads(line)
+                        response_part = body.get("response")
+                        yield response_part
+                        if body.get("done"):
+                            break
```

Comment:
Same as previous comments, maybe refactor with a `AsyncSession` context manager?

Reply:
I remember seeing this changes and now they are gone and the duplicated logic is back in.